src/profile-logic/process-profile.js

```diff
@@ -954,13 +954,26 @@ function _processMarkerPayload(
   }
 }
 
+function _timeColumnToTimeDeltas(time: number[]): number[] {
+  const NS_PER_MS = 1000000;
+  const timeDeltas = new Array(time.length);
+  let prevTimeNs = 0;
+  for (let i = 0; i < time.length; i++) {
+    const currentTimeNs = Math.round(time[i] * NS_PER_MS);
```

Comment:
I'm a bit worried that by multiplying by 1_000_000 we get closer to the max safe integer.
(although this seems to be ~2501 hours so it's probably OK);

A comment explaining why you're doing this dance with ns and round would be good!

Reply:
Good point, I'll add a comment. I added this because otherwise a lot of the numbers in the snapshot tests became longer.

Hrmmm and you're right about max safe integer being a bit of a problem. For example, `Math.round(9007199254.741003 * 1000000) / 1000000  === 9007199254.741005` and `9007199254.741005 !== 9007199254.741003`. I don't think it matters in practice but I can't prove it at the moment.

Reply:
I'm adding the following comment:

```js
  // For each timestamp in the time series, compute the delta to the previous
  // timestamp. The implicit initial timestamp is zero.
  //
  // Timestamps are in milliseconds. To compute the deltas, we first convert each
  // timestamp to integer nanoseconds. Then we subtract those nanosecond timestamps
  // and converting the delta to milliseconds again. We do this dance so that
  // the deltas have a "compact" stringified representation. Otherwise,
  // converting to deltas could easily increase the JSON size:
  // For example, 252.728334 - 240.520375 === 12.207958999999988.
```